blockchain/backup/NFTtheWorld.sol

```diff
@@ -183,24 +196,98 @@ contract NFTtheWorld {
         availableNFTsCount[_dropHash] = availableNFTs[_dropHash].length;
     }
 
-    function getNFTIndex(uint256 _nftHash, uint256 _dropHash)
+    // Helper function to shuffle a list
+    function shuffle(uint256 _dropHash) internal {
+        for (uint256 i = 0; i < availableNFTs[_dropHash].length; i++) {
+            uint256 n = i +
+                (uint256(keccak256(abi.encodePacked(block.timestamp))) %
+                    (availableNFTs[_dropHash].length - i));
+            string memory temp = availableNFTs[_dropHash][n];
+            availableNFTs[_dropHash][n] = availableNFTs[_dropHash][i];
+            availableNFTs[_dropHash][i] = temp;
+        }
+    }
+
+    function getNFTIndex(string memory _uri, uint256 _dropHash)
         internal
         view
         returns (uint256 index)
     {
         NFTOwnership[] memory nfts = nftOwnerships[_dropHash];
         for (uint256 i = 0; i < nfts.length; i++) {
-            if (nfts[i].nftId == _nftHash) {
+            if (compareStrings(nfts[i].uri, _uri)) {
                 return i;
             }
         }
     }
 
+    function compareStrings(string memory a, string memory b)
+        private
+        pure
+        returns (bool)
+    {
+        return (keccak256(abi.encodePacked((a))) ==
+            keccak256(abi.encodePacked((b))));
+    }
+
     function getDropTime(uint256 _dropHash)
         public
         view
         returns (uint256 dropTime)
     {
         return nftOwnerships[_dropHash][0].dropTime;
     }
+
+    // Modifier to check if msg.sender is elligible
+    modifier onlyByAdmins() {
+        require(
+            adminRights[msg.sender] == true,
+            "Your are not elligible to perform this action."
+        );
+        _;
+    }
+
+    function addToAdmins(address payable _addressToAdd) public onlyByAdmins {
+        adminRights[_addressToAdd] = true;
+    }
+
+    //Restriction to not remove oneself from admin list has been commented out for testing reasons, but generally is desired
+    function removeFromAdmins(address payable _addressToRemove)
+        public
+        onlyByAdmins
+    {
+        adminRights[_addressToRemove] = false;
```

Comment:
As written in the other PR I reviewed, I think it would be a good idea to forbid that admins can remove themselves. But for now it probably makes testing a lot easer if this is not part of the function. But something to maybe keep in mind :D

Reply:
oh no, when Jannis and I merged today we forgot this line, he already implemented it. Will add again